glean-core/src/database/mod.rs

```diff
@@ -537,13 +543,42 @@ impl Database {
             .write()
             .expect("Can't access app lifetime data as writable")
             .clear();
+    }
 
+    /// Persist ping_lifetime_data to disk and clear it.
+    ///
+    /// Does nothing in case there is nothing to persist.
+    ///
+    /// ## Panics
+    ///
+    /// * This function will **not** panic on database errors.
+    pub fn persist_ping_lifetime_data(&self) -> Result<()> {
         if let Some(ping_lifetime_data) = &self.ping_lifetime_data {
+            {
```

Comment:
Why do we need the extra scope/curly brackets here?

Reply:
Because `ping_lifetime_data` holds a lock while reading the data. The brackets limit the scope of that lock, so that we can clear the data after reading it.

Reply:
Thanks for explaining, that makes sense!